Sources/MultiSets/CountedSet/CountedSet+Sequence.swift

```diff
@@ -0,0 +1,41 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Collections open source project
+//
+// Copyright (c) 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+extension CountedSet: Sequence {
+  /// Returns an iterator over the elements of this collection.
+  ///
+  /// - Complexity: O(1)
+  /// - Remark: A type-erased wrapper is used instead of an opaque type purely
+  /// to preserve compatibility with older versions of Swift.
+  @inlinable
+  public func makeIterator() -> AnyIterator<Element> {
```

Comment:
Yeah, I figured as much. While I couldn’t see any downside to using `AnyIterator`, the fact that nothing else seemed to was a bad sign.

I don’t suppose you could explain why that’s frowned upon here? As the comment notes, I’d have used an opaque type if I was allowed. Is there some benefit to making a dedicated type?

Reply:
Oh, it's a performance issue. Like `AnySequence`, `AnyCollection` and `AnyHashable`, the type-erased iterator needs to box things up into a class (or a closure, I can't remember & I'm too lazy to check 😈) to get rid of the original types and then dispatch dynamically to the underlying `next()` method, and this tends to make things much slower than usual.

`AnyIterator` isn't a bad construct -- it does have its use cases. But it's almost always better to design things so that we don't need to use it, especially in cases where performance is important. (Which is definitely the case for this package.)

Reply:
(`AnyIterator` is really nice in that it hides the actual iterator type we're using so that we remain able to change it later. I expect `some IteratorProtocol` will at some point let us do the same thing without the performance impact -- unfortunately currently we can't use it as we can't constrain the Element type of such.)